Sources/Identity/CustomerInfoManager.swift

```diff
@@ -322,24 +322,36 @@ private extension CustomerInfoManager {
                 let transactions = await self.transactionFetcher.unfinishedVerifiedTransactions
 
                 if !transactions.isEmpty {
-                    var results: [Result<CustomerInfo, BackendError>] = []
-                    let storefront = await Storefront.currentStorefront
-
                     Logger.debug(
                         Strings.customerInfo.posting_transactions_in_lieu_of_fetching_customerinfo(transactions)
                     )
 
-                    for transaction in transactions {
-                        results.append(
-                            await self.transactionPoster.handlePurchasedTransaction(
-                                transaction,
-                                data: .init(appUserID: appUserID,
-                                            presentedOfferingID: nil,
-                                            unsyncedAttributes: [:],
-                                            storefront: storefront,
-                                            source: Self.sourceForUnfinishedTransaction)
-                            )
-                        )
+                    let storefront = await Storefront.currentStorefront
+
+                    // Process all transactions in parallel
+                    let results: [Result<CustomerInfo, BackendError>] = await withTaskGroup(
+                        of: Result<CustomerInfo, BackendError>.self
+                    ) { group in
+                        for transaction in transactions {
+                            group.addTask {
+                                return await self.transactionPoster.handlePurchasedTransaction(
+                                    transaction,
+                                    data: .init(appUserID: appUserID,
+                                                presentedOfferingID: nil,
+                                                unsyncedAttributes: [:],
+                                                storefront: storefront,
+                                                source: Self.sourceForUnfinishedTransaction)
+                                )
+                            }
+                        }
+
+                        var results: [Result<CustomerInfo, BackendError>] = []
+
+                        for await result in group {
+                            results.append(result)
+                        }
+
+                        return results
                     }
 
                     // Any of the POST receipt operations will have posted the same receipt contents
```

Comment:
This code (that I can barely comment on) says that it fetches the "last known success if available".

Do these transactions not being posted in sequential order affect what the `lastSuccess` is? Not expert with tasks groups so just wanted to make sure 😁

Reply:
thinking about this more... We don't need to even create N post receipts for this, right? Like, we could just post once and assume that it'll get everything we need. As in: 

if unfinishedTransactions not empty
    customerInfo = post first unfinished transaction
    return customerInfo

Reply:
We've talked about this in the past. The assumption is that it doesn't matter because we'll de-dup. But by doing this, we ensure that we finish all transactions. Otherwise the next time you call `getCustomerInfo` the same thing will happen with N-1 transactions.